core/src/main/java/org/apache/carbondata/core/datastore/page/statistics/PrimitivePageStatsCollector.java

```diff
@@ -233,20 +233,18 @@ public void update(long value) {
 
   /**
    * Return number of digit after decimal point
-   * TODO: it operation is costly, optimize for performance
    */
   private int getDecimalCount(double value) {
     int decimalPlaces = 0;
     try {
-      String strValue = BigDecimal.valueOf(Math.abs(value)).toPlainString();
-      int integerPlaces = strValue.indexOf('.');
-      if (-1 != integerPlaces) {
-        decimalPlaces = strValue.length() - integerPlaces - 1;
+      BigDecimal decimalValue = BigDecimal.valueOf(value);
```

Comment:
better to write code to implement it.
not required to use BigDecimal.

Reply:
Actually double will not always be like `xx.yyy`, it will be having `exponent` also. 
So, Bigdecimal already converts to string and do that logic. 

May be next version we can reduce further by removing big decimal and copying API inside big decimal to do that.

Reply:
Also we need something without converting to string